reactioncommerce/api-plugin-files
# feat: add source code and enable semantic-release

This plugin replaces the internal `files` service.

This PR:
- copies the source code from `reaction`
- updates the licence to GPL-3
- removes testing related configuration since the plugin has no tests
- enables semantic-release

Once this PR is merged and this plugin is published to `npm`, there will be a related PR in `reaction` to swap out this plugin for the existing version.

## Testing

- see that the plugin is loaded upon startup in Reaction (use the `systemInformation` query)
- use the `updateMediaRecordPriority` mutation and verify it's working
- upload a file through Admin and verify it's working (I haven't tested this one due to several problems with Admin on my local environment, so please make sure you do)

Issue https://github.com/reactioncommerce/reaction/issues/6172

Signed-off-by: Maria Paktiti <maria.paktiti@gmail.com>
